rmf_traffic/src/rmf_traffic/schedule/ViewerInternal.hpp

```diff
@@ -35,7 +35,7 @@ class Viewer::View::Implementation
     ParticipantId participant;
     RouteId route_id;
     ConstRoutePtr route;
-    std::shared_ptr<const ParticipantDescription> description;
+    std::shared_ptr<std::shared_ptr<const ParticipantDescription>> description;
```

Comment:
This should actually remain as `std::shared_ptr<const ParticipantDescription>` since its purpose is to maintain the lifetime of the specific immutable `const ParticipantDescription` that the view is supposed to provide. If the Viewer receives a new `ParticipantDescription` then this value could suddenly change, which breaks the immutability guarantee that `View` is meant to have.

Reply:
Fixed in e855ae6.